File: 242-valid-anagram/valid-anagram.py
class Solution:
    def wordToDict(self, word):
        d = {}
        for letter in word:
            d[letter] = d.get(letter, 0) + 1
        return d
    def isAnagram(self, s: str, t: str) -> bool:
        return self.wordToDict(s) == self.wordToDict(t)

File: 242-valid-anagram/valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length)
            return false;
        
        Dictionary<char, int> ds = WordToDict(s);
        Dictionary<char, int> dt = WordToDict(t);
        foreach(char key in ds.Keys)
        {
            if (dt.ContainsKey(key) == false
                || dt[key] != ds[key])
                return false;
        }

        return true;
    }

    private Dictionary<char, int> WordToDict(string word) {
        Dictionary<char, int> result = new Dictionary<char, int>();

        foreach(char c in word)
            result[c] = result.GetValueOrDefault(c, 0) + 1;

        return result;
    }
}